Algorithm.md

```diff
@@ -0,0 +1,30 @@
+1. Fetch available versions for each dependency
+2. If a project is already in the graph, intersect the new requirement with the one already present
+3. Select the highest [precedence] version which satisfies the intersected requirement (or only requirement, if this is the first insertion)
+  * This always has to be O(n) over the list of versions, because custom requirements mean that a previously failing version might become a candidate again
+4. Collect all child dependencies from that graph depth, repeat from step 1
+
+### In event of a conflict
+
+Given a project already in the graph with requirement X, and about to be re-added to the graph with requirement Y, where X and Y conflict:
+
+1. Find the _dependent_ project which changed the requirement into X (either from nil, or from a previously looser requirement)
+2. Select the next best version of that dependent project and attempt to finish resolving
+3. In the event that no other versions satisfy the requirements applied to the dependent, we have two choices:
+  - GOTO step 1 for the dependent project, or
+  - Find a further back time where the originally conflicting dependency (the one with requirement X) was added (e.g., with requirement W), and try replacing the dependent which added it at _that_ level instead
```

Comment:
I've ordered the hardcover version of the book, because that does sound interesting. Thanks for the recommendation!

In this particular context, though, is there something about the behavioral specification that you think should change, or are you just suggesting implementation considerations?